config.toml

```diff
@@ -0,0 +1,221 @@
+# In current form it doesn't work with anything that's escaped, so no `\w` etc.
+# In this case, "Carrier Name" contains the "User Number" at the very end. The same issue is not present in Voice_RM.
+encode_regex = [['^.* (?P<user_number>[0-9]+)$', 'Carrier Name']]
+
+[Rogers.SMS]
+file_mask = 'ALL_CALLS-SMS\.'
+clear_columns = ['Subscriber ID 1', 'Subscriber ID 2', 'Subscriber ID 3', 'Subscriber ID 4']
+encode_columns = ['BAN', 'Invoice No.', 'User Name', 'User Number']
+external_header_file = 'Header-SMS.txt'
+# Encode value in `Called Number` if it's not an asterisk.
+encode_conditional = [['Called Number', 'Called Number', '!=', '*']]
+
+[Rogers.Voice]
+file_mask = 'ALL_CALLS-Voice\.'
+clear_columns = ['Subscriber ID 1', 'Subscriber ID 2', 'Subscriber ID 3', 'Subscriber ID 4']
+encode_columns = ['BAN', 'Invoice No.', 'User Name', 'User Number']
+external_header_file = 'Header-Voice.txt'
+encode_conditional = [['Called Number', 'Called Number', '!=', '*']]
+
+[Rogers.Voice_RM]
+file_mask = 'ALL_CALLS-VoiceRM\.'
+clear_columns = ['Subscriber ID 1', 'Subscriber ID 2', 'Subscriber ID 3', 'Subscriber ID 4']
+encode_columns = ['BAN', 'Invoice No.', 'User Name', 'User Number']
+external_header_file = 'Header-Voice.txt'
+encode_conditional = [['Called Number', 'Called Number', '!=', '*']]
+# Note that GPRS_RM file has this issue where "User Number" is placed in "Carrier Name" field. This issue
+# is not present in the Voice_RM file, in which only the country and the carrier are listed.
+
+
+[Telus]
+[Telus.common]
+carrier = 'Telus'
+config_class = 'raw-regex-config'
+encoding = 'iso-8859-1'
+regex_groups = [
+    # Header.
+    'Client No:\t"(?P<client_number>[0-9]+)"',
+    'Unique Invoice No:\t"(?P<invoice_number>[0-9]+)"',
+
+    # Body.
+    '^"Account Number (?P<account_number>[0-9]{8})"',
+    # Keeping the actual device information, could be useful for parsing out device types.
+    '^"Number/Device Information"\t"(?P<number>[0-9]{10})/(.*)"',
+    '^"C:"\t"(?P<phone_number>.*)"',
+    '^"Subscriber Name:"\t"(?P<subscriber_name>.*)"',
+    '^"Additional line user name:"\t"(?P<additional_user_name>.*)"',
+    '^"Sublevel:"\t"(?P<sublevel_name>.*)"',
+    '^"Reference ([0-9]+):"\t"(?P<reference_name>.*)"',
+
+    # Weird tables.
+    '^Department = (?P<department>.*)$',
+    '^Cost Center = (?P<department>.*)$',
+
+    # Tables.
+    # DEW Report starts with Name and then Unit Number.
+    '^"(?P<name>[^"]+)"\t"(?P<number>[0-9]{10})"\t',
+    # Group Summary Reports. It consists of:
+    # 2x subgroup, 2x reference, a single character for "prod type", then unit number and two user names,
+    # all separated by tab characters. From all that, only "prod type", number and first name cannot be empty.
+    '^(?P<sublevel_1>[^\t]*)\t(?P<sublevel_2>[^\t]*)\t(?P<ref_1>[^\t]*)\t(?P<ref_2>[^\t]*)\t([a-zA-Z])\t(?P<number>[0-9]{10})\t(?P<name_1>[^\t]+)\t(?P<name_2>[^\t]*)\t',
```

Comment:
this seems very fragile

wouldn't it be better to identify all "secret" strings, like user name or phone numbers, then go from the longest string to shortest and run replace on WHOLE file contents?

that way as long as username is represented in same way on all files analyzed, once we identify it in single place, we can censor it everywhere - even in tricky places like this one